dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Functions/GenerateProviderExportFunction.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.Providerportal.FindAnApprenticeship.Interfaces.Services;
+using Dfc.Providerportal.FindAnApprenticeship.Models;
+using Dfc.Providerportal.FindAnApprenticeship.Storage;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Dfc.Providerportal.FindAnApprenticeship.Functions
+{
+    public class GenerateProviderExportFunction
+    {
+        private readonly IApprenticeshipService _apprenticeshipService;
+        private readonly IBlobStorageClient _blobStorageClient;
+
+        public GenerateProviderExportFunction(IApprenticeshipService apprenticeshipService, IBlobStorageClient blobStorageClient)
+        {
+            _apprenticeshipService = apprenticeshipService ?? throw new ArgumentNullException(nameof(apprenticeshipService));
+            _blobStorageClient = blobStorageClient ?? throw new ArgumentNullException(nameof(blobStorageClient));
+        }
+
+        [FunctionName("GenerateProviderExport")]
+        public async Task Run([TimerTrigger("%GenerateProviderExportSchedule%")]TimerInfo timer, ILogger log, CancellationToken ct)
+        {
+            var exportKey = ExportKey.FromUtcNow();
+
+            log.LogInformation($"Started generation of {{{nameof(exportKey)}}}.", exportKey);
+
+            string export = null;
+            try
+            {
+                var apprenticeships = (List<Apprenticeship>)await _apprenticeshipService.GetLiveApprenticeships();
+
+                export = JsonConvert.SerializeObject(
+                    (await _apprenticeshipService.ApprenticeshipsToDasProviders(apprenticeships)).Where(r => r.Success).Select(r => r.Result),
+                    new JsonSerializerSettings { ContractResolver = new CamelCasePropertyNamesContractResolver() });
+
+                log.LogInformation($"Completed generation of {{{nameof(exportKey)}}}.", exportKey);
+            }
+            catch (Exception ex)
+            {
+                log.LogError(ex, $"Failed to generate {{{nameof(exportKey)}}}.", exportKey);
```

Comment:
I think you need a `return;` here to stop the upload processing

Reply:
Ooooh well spotted. I did have a throw in there.